SpokestackTests/NLUTensorflowSlotParserTest.swift

```diff
@@ -26,45 +26,52 @@ class NLUTensorflowSlotParserTest: XCTestCase {
+        let et5 = EncodedTokens(tokensByWhitespace: ["4238341745"], encodedTokensByWhitespaceIndex: [0, 0, 0, 0, 0], encodedTokens: [0])
+        let parsedDigitsNumeric = try! parser.parse(tags: taggedInputPhoneNumeric, intent: metadata!.intents.filter({ $0.name == "inform.phone_number" }).first!, encoder: encoder!, encodedTokens: et5)
         XCTAssertEqual(parsedDigitsNumeric["phone_number"]!.value as! String, "4238341745")
 
-        let taggedInputPhoneCardinal = zip(["b_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number"],["4", "second", "three", "eighth", "three", "four", "one", "seven", "four", "five"])
-        let parsedDigitsCardinal = try! parser.parse(taggedInput: taggedInputPhoneCardinal, intent: metadata!.intents.filter({ $0.name == "inform.phone_number" }).first!, encoder: encoder!)
+        let taggedInputPhoneCardinal = ["b_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number"]
+        let et6 = EncodedTokens(tokensByWhitespace: ["4", "second", "three", "eighth", "three", "four", "one", "seven", "four", "five"], encodedTokensByWhitespaceIndex: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], encodedTokens: [0])
+        let parsedDigitsCardinal = try! parser.parse(tags: taggedInputPhoneCardinal, intent: metadata!.intents.filter({ $0.name == "inform.phone_number" }).first!, encoder: encoder!, encodedTokens: et6)
         XCTAssertEqual(parsedDigitsCardinal["phone_number"]!.value as! String, "4238341745")
     }
     
     func testParseEntity() {
-        let taggedInputEntity = zip(["b_epynonymous", "i_epynonymous"],["dead", "beef"])
-        let parsedEntity = try! parser.parse(taggedInput: taggedInputEntity, intent: metadata!.intents.filter({ $0.name == "identify" }).first!, encoder: encoder!)
+        let taggedInputEntity = ["b_epynonymous", "i_epynonymous"]
+        let et1 = EncodedTokens(tokensByWhitespace: ["dead", "beef"], encodedTokensByWhitespaceIndex: [0, 1], encodedTokens: [0])
+        let parsedEntity = try! parser.parse(tags: taggedInputEntity, intent: metadata!.intents.filter({ $0.name == "identify" }).first!, encoder: encoder!, encodedTokens: et1)
         XCTAssertEqual(parsedEntity["epynonymous"]!.value as! String, "dead beef")
+        
+        let taggedInputEntityO = ["o", "b_epynonymous", "i_epynonymous", "o", "o", "b_epynonymous", "o"]
+        let etO = EncodedTokens(tokensByWhitespace: ["when", "dead", "beef", "appears", "in", "debug"], encodedTokensByWhitespaceIndex: [0, 1, 2, 3, 4, 5], encodedTokens: [0])
+        let parsedEntityO = try! parser.parse(tags: taggedInputEntityO, intent: metadata!.intents.filter({ $0.name == "identify" }).first!, encoder: encoder!, encodedTokens: etO)
+        XCTAssertEqual(parsedEntityO["epynonymous"]!.value as! String, "dead beef debug")
```

Comment:
Interesting. This would be a strange thing to get from the model. I think the Android version would just give you `"debug"` as the value here instead of concatenating the strings. We don't currently support or claim to support list/array values for slots.

Reply:
oh interesting, i thought that entities constituted all of a single tag, not just a range.

Reply:
i…don't think the discrepancy is problematic, but defer to you on that.